### PR TITLE
Use config component in settings component

### PR DESCRIPTION
**Pull request by GustavoCaso**
Related Jira tickets: https://datadoghq.atlassian.net/browse/ASCII-1464, https://datadoghq.atlassian.net/browse/ASCII-1465

### What does this PR do?

Remove the use of `pkg/config` in the Settings component. 

The idea is to not use the `pkg/config` but rather our config component. Since in all cases of the setting the `Set` and `Get` reach the config object. 

To refactor the code I had to get access to the `config.Component` inside the Settings component. 

### What is the reason I to use `settings.Params` to pass the config object rather than dependencies?

Since the system probe agent uses their dedicated component `sysprobeconfig.Component`. The system probe app uses both `config.Component` and `sysprobeconfig.Component` Fx would not be able to resolve which component interface to use. To overcome that I created the `Setting.Params` which allow us to be more deliberate to pass the right config component to use depending on the agent. 

Also, we remove the old `pkg/setting/runtime_settings.go` 🔥 

### Motivation

Reduce usage of `pkg/config` 

### Describe how to test/QA your changes

Test that the config CLI commands for each of the agents work as expected

- `config`
- `list-runtime`
- `get <config>`
- `set <config> <value>`

For remote configuration validate that the changing the log level via UI works as expected

### CODE REVIEW

**Comment by GustavoCaso** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable RUN_E2E_TESTS=on --variable RUN_UNIT_TESTS=on

File: comp/core/settings/component.go [hunk 1]
```diff
@@ -32,13 +32,18 @@ type RuntimeSettingResponse struct {
...
 }
 
-// Settings define the runtime settings the component would understand
-type Settings map[string]RuntimeSetting
+// Params that the settings component need
+type Params struct {
+	// Settings define the runtime settings the component would understand
+	Settings map[string]RuntimeSetting
+	// The config object in which the settings would apply operations
+	Config config.Component
```
Comment (hush-hush) [inline comp/core/settings/component.go:40]: It's often hard to know which config we're working with for runtime settings (core, system-probe, security ...).

What about having a dedicated field for each ?

**Review by hush-hush — APPROVED** [review]
Added a quick question, but looks great overall.

**Review by paulcacheux — APPROVED** [review]

File: comp/core/settings/component.go (hunk 1 above)
Comment (GustavoCaso) [reply comp/core/settings/component.go:40]: That is fair point. 

As discussed IRL. Today each agent process handles their own configuration, so each setting component needs a single config component to interact. 

In the future when settings configuration is only handled via the core agent we could store each of the config components and change the right config based on the config prefix. Ex: `datadod-agent config set apm_config.log_level` This command would just modify the apm process setting

**Review by hush-hush — APPROVED** [review]

**Review by clamoriniere — APPROVED** [review]
/lgtm for @DataDog/container-integrations  team files

**Review by moth-a — APPROVED** [review]
RC changes lgtm 👍

**Review by robertjli — APPROVED** [review]

**Review by brycekahle — APPROVED** [review]

**Review by NouemanKHAL — APPROVED** [review]

**Comment by GustavoCaso** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable RUN_E2E_TESTS=on --variable RUN_UNIT_TESTS=on

**Pull request merged**